resources/ecr-scan-check.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+
+import argparse
+import boto3
+import json
+import os
+import sys
+import time
+
+
+def logoutput(level, message):
+    colours = {
+        "Info": "\033[1;34m",
+        "Error": "\033[0;31m",
+        "Success": "\033[0;32m",
+        "end": "\033[0m",
+    }
+
+    string = "{}{}:{} {}".format(colours[level], level, colours["end"], message)
+    print(string)
+
+
+def get_severities(min_severity):
+    severities = ["CRITICAL", "HIGH", "MEDIUM", "LOW", "INFORMATIONAL"]
+
+    if min_severity == "ANY":
+        selected_severities = severities
+    else:
+        selected_severities = []
+        for severity in severities:
+            selected_severities.append(severity)
+            if severity == min_severity:
+                break
+
+    return selected_severities
+
+
+def ecr_opensession():
+    logoutput("Info", "Logging in to ECR")
+    try:
+        session = boto3.Session(
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            region_name=os.environ.get("AWS_REGION"),
+        )
+        ecr_client = session.client("ecr")
+
+    except session.exceptions.UnauthorizedException as err:
+        logoutput("Error", "Unable to authorise request. Invalid credentials")
+        sys.exit(1)
+
+    return ecr_client
+
+
+def ecr_describescanfindings(ecr_client, imagerepo, imagetag):
+    logoutput("Info", "Getting scan results...")
+    try:
+        response = ecr_client.describe_image_scan_findings(
+            repositoryName=imagerepo,
+            imageId={
+                "imageTag": imagetag,
+            },
+        )
+
+        logoutput("Info", "Image digest: {}".format(response["imageId"]["imageDigest"]))
+
+    except ecr_client.exceptions.ImageNotFoundException as err:
+        logoutput(
+            "Error", "No results found for image " + imagerepo + " with tag " + imagetag
+        )
+        sys.exit(1)
+
+    except ecr_client.exceptions.RepositoryNotFoundException as err:
+        logoutput("Error", "No repository found with name " + imagerepo)
+        sys.exit(1)
+
+    return response
+
+
+def parse_scanresults(response, severities):
+    logoutput("Info", "Severities included: " + ",".join(severities))
+    vulnerabilities = False
+    if len(response["imageScanFindings"]["findingSeverityCounts"]) > 0:
+        for severity in response["imageScanFindings"]["findingSeverityCounts"]:
+            if severity in severities:
+                vulnerabilities = True
+                count = response["imageScanFindings"]["findingSeverityCounts"][severity]
+                logoutput("Error", "{}: {}".format(severity, count))
+
+    return vulnerabilities
+
+
+def generate_reporturl(ecr_scanfindings, region):
+    report_url = "https://{region}.console.aws.amazon.com/ecr/repositories/private/{registry}/{repository}/image/{digest}/scan-results/?region={region}".format(
```

Comment:
Not really avoidable, unfortunately, as the string is constructing an AWS console URL.
The string could be broken up in to constituent parts but that would lead to lack of clarity.